config/rbac/kustomization.yaml

```diff
@@ -7,12 +7,12 @@ resources:
 - service_account.yaml
 - role.yaml
 - role_binding.yaml
-- leader_election_role.yaml
-- leader_election_role_binding.yaml
+#- leader_election_role.yaml
+#- leader_election_role_binding.yaml
 # Comment the following 4 lines if you want to disable
 # the auth proxy (https://github.com/brancz/kube-rbac-proxy)
 # which protects your /metrics endpoint.
-- auth_proxy_service.yaml
-- auth_proxy_role.yaml
-- auth_proxy_role_binding.yaml
-- auth_proxy_client_clusterrole.yaml
+#- auth_proxy_service.yaml
```

Comment:
I think we will ultimately want the metrics auth proxy to ensure metrics are scraped over HTTPS. Any concern including that now?

Reply:
No concern, just forgot we start metrics server using the controller runtime's manager. Will add it to this PR.

Reply:
Added `kube-rbac-proxy` sidecar with all needed rbac:
```bash
$ oc get pods
NAME                                     READY   STATUS    RESTARTS   AGE
external-dns-operator-7557bd6f49-m2rk8   2/2     Running   0          6m41s

$ oc logs external-dns-operator-7557bd6f49-m2rk8 -c kube-rbac-proxy
I0713 08:59:58.432187       1 main.go:190] Valid token audiences: 
I0713 08:59:58.432283       1 main.go:262] Generating self signed cert as no cert is provided
I0713 08:59:58.807667       1 main.go:311] Starting TCP socket on 0.0.0.0:8443
I0713 08:59:58.807918       1 main.go:318] Listening securely on 0.0.0.0:8443
```